### PR TITLE
Editable edges

### PR DESCRIPTION
**Pull request by jbicker**

### CODE REVIEW

File: client/src/features/select/initializer.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import { ActionHandlerRegistry, IActionHandler, IActionHandlerInitializer } from "../../base/actions/action-handler"
+import { Action } from "../../base/actions/action"
+import { ICommand } from "../../base/commands/command"
+import { SelectAction, SelectCommand } from "./select"
+import { ActivateEditModeAction } from "../edit/edit"
+
+
+class ActivateEditModeHandler implements IActionHandler {
+    handle(action: SelectAction): void | ICommand | Action {
+        console.log("bla action ", action)
```
Comment (JanKoehnlein) [inline client/src/features/select/initializer.ts]: please remove

File: client/src/graph/sgraph.ts [hunk 2]
```diff
@@ -114,7 +117,17 @@ export class SEdge extends SChildElement implements Fadeable {
     }
 
     hasFeature(feature: symbol): boolean {
-        return feature === fadeFeature
+        return feature === fadeFeature || feature === selectFeature || feature === editFeature
+    }
+}
+
+export class SControlPoint extends SChildElement implements Fadeable, Selectable, Locateable {
```
Comment (JanKoehnlein) [inline client/src/graph/sgraph.ts]: Could we call it SRoutingPoint?

File: client/src/features/select/initializer.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+import { injectable } from "inversify"
+import { ActionHandlerRegistry, IActionHandler, IActionHandlerInitializer } from "../../base/actions/action-handler"
+import { Action } from "../../base/actions/action"
+import { ICommand } from "../../base/commands/command"
+import { SelectAction, SelectCommand } from "./select"
+import { ActivateEditModeAction } from "../edit/edit"
+
+
+class ActivateEditModeHandler implements IActionHandler {
```
Comment (JanKoehnlein) [inline client/src/features/select/initializer.ts]: Shouldn't that class reside in `features/edit` instead?

File: client/src/features/edit/edit.ts [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+import { centerOfLine, Point } from "../../utils/geometry"
+import { SelectAction } from "../select/select"
+import { findParent } from "../../base/model/smodel-utils"
+import { ElementMove } from "../move/move"
+import { KeyListener } from "../../base/views/key-tool"
+
+export class ShowControlPointsAction implements Action {
+    kind: string = ShowControlPointsCommand.KIND
+
+    constructor(public priviousAction?: Action) {
```
Comment (JanKoehnlein) [inline client/src/features/edit/edit.ts]: What is the `previousAction` good for?
Have you considered using a `MergeableCommand`?

File: client/src/utils/geometry.ts [hunk 5]
```diff
@@ -111,6 +111,16 @@ export function center(b: Bounds): Point {
     }
 }
 
+export function centerOfLine(s: Point, e: Point): Point {
+    const b: Bounds = {
+        x: s.x > e.x ? e.x : s.x,
```
Comment (JanKoehnlein) [inline client/src/utils/geometry.ts]: `Math.min()`?

File: client/src/features/edit/edit.ts [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+export class ShowControlPointsCommand implements Command {
+    static KIND: string = "controlPointsVisible"
+
+    constructor(public action: ShowControlPointsAction) {
+    }
+
+    execute(context: CommandExecutionContext): CommandResult {
+        const sModelRoot = context.root
+
+        const createControlPoint = (cptype: string, id: string, position: Point) => {
```
Comment (JanKoehnlein) [inline client/src/features/edit/edit.ts]: Please convert to protected method. It's much easier to reuse/customize for the users

File: client/src/features/edit/edit.ts [hunk 7]
```diff
@@ -0,0 +1,253 @@
...
+            const sControlPoint = new SControlPoint()
+            sControlPoint.type = cptype
+            sControlPoint.id = id
+            sControlPoint.position = position
+            return sControlPoint
+        }
+
+        const showControlPoint = (editTarget: Editable) => {
```
Comment (JanKoehnlein) [inline client/src/features/edit/edit.ts]: Please convert to protected method. It's much easier to reuse/customize for the users

**Comment by spoenemann** [comment]
I rebased this work on the current head and pushed the result to https://github.com/theia-ide/sprotty/tree/EditableEdges2. I'll try to make that work with the current state and create a new PR.

**Pull request closed without merging**